WalletWasabi/Tor/Socks5/Pool/TorHttpPool.cs

```diff
@@ -259,6 +272,12 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, ICi
 						connectionToDispose.MarkAsToDispose();
 					}
 
+					// If our request failed but other requests seem to work, try a different circuit.
+					if (!attemptSuccessful && TorDoesntWorkSince is null)
+					{
+						namedCircuit.IncrementIsolationId();
+					}
```

Comment:
Yes. It looks even better, because you're using `TorDoesntWorkSince` null check that in theory should prevent circuit exhaustion - if there's such a thing. I hope it also works.